build-scripts/packageExtension.js

```diff
@@ -0,0 +1,118 @@
+        console.log('Processing task ' + taskName)
+
+        if(!unprocessFolders.every((folderName) => { return folderName !== taskName})) {
+            console.log('Skpping task ' + taskName)
+            return
+        }
+
+        var taskBuildFolder = path.join(outBuildTasks, taskName)
+        var taskPackageFolder = path.join(outPackageTasks, taskName)
+        fs.mkdirpSync(taskPackageFolder)
+
+        var taskDef = require(path.join(taskBuildFolder, 'task.json'))
+        if (!taskDef.execution.hasOwnProperty('Node')) {
+            console.log('Copying non-node task ' + taskName)
+
+            fs.copySync(taskBuildFolder, taskPackageFolder)
+            return
+        }
+        shell.cd(taskBuildFolder)
+        fs.copySync(path.join(taskBuildFolder, 'task.json'), path.join(taskPackageFolder, 'task.json'), {overwrite: true})
+        fs.copySync(path.join(taskBuildFolder, 'task.loc.json'), path.join(taskPackageFolder, 'task.loc.json'), {overwrite: true})
+        fs.copySync(path.join(taskBuildFolder, 'package.json'), path.join(taskPackageFolder, 'package.json'), {overwrite: true})
+        fs.copySync(path.join(taskBuildFolder, 'icon.png'), path.join(taskPackageFolder, 'icon.png'), {overwrite: true})
+        fs.copySync(path.join(taskBuildFolder, 'Strings'), path.join(taskPackageFolder, 'Strings'), {overwrite: true})
+
+        console.log('packing node-based task')
+        var webpackConfig = path.join(repoRoot, 'webpack.config.js')
+        var webpackCmd = 'webpack ' 
+                        + '--config ' + webpackConfig + ' '
+                        + taskName + '.js '
+                        + '--output-path ' + path.join(taskPackageFolder) + ' '
+                        + '--output-filename ' + taskName + '.js' + ' '
+        console.log(webpackCmd)
+        try {
+            ncp.execSync(webpackCmd, {stdio: 'pipe'})
+        }
+        catch (err) {
+            console.error(err.output ? err.output.toString() : err.message)
+            process.exit(1)
+        }
+
+        shell.cd(taskPackageFolder)
+        var npmCmd = 'npm install vsts-task-lib --only=production'
+        try {
+            output = ncp.execSync(npmCmd, {stdio: 'pipe'})
+            console.log(output)
+        }
+        catch (err) {
+            console.error(err.output ? err.output.toString() : err.message)
+            process.exit(1)
+        }
+
+        shell.cd(repoRoot)
+    })
+
+    console.log('Creating deployment vsix')
+    var tfxcmd = 'tfx extension create --root ' + outPackage + ' --output-path ' + outPackage + ' --manifests ' + path.join(outPackage, manifestFile)
+    if (options.publisher) {
+        tfxcmd += ' --publisher ' + options.publisher
+    }
+
+    console.log('Packaging with:' + tfxcmd)
+
+    ncp.execSync(tfxcmd, {stdio: 'pipe'})
+
+    console.log('Packaging successful')
+}
+
+console.time(timeMessage)
+var options = process.argv.slice(2)
+if(options.length > 0 && options[0].startsWith('publisher')) {
```

Comment:
Make an exact match (this would work with `publisherx=123`)

Reply:
I disagree, this is already a hacky way to parse arguments and there is no harm in allowing that in a build